cmake/tribits/common_tpls/FindTPLHDF5.cmake

```diff
@@ -16,10 +18,13 @@ if (Netcdf_ALLOW_MODERN)
       HDF5
       INNER_FIND_PACKAGE_NAME  HDF5
       IMPORTED_TARGETS_FOR_ALL_LIBS   ${HDF5_EXPORT_LIBRARIES})
+    set(INTERNAL_IS_MODERN TRUE)
   endif()
 
 endif()
 
+set(HDF5_FOUND_MODERN_CONFIG_FILE ${INTERNAL_IS_MODERN} CACHE BOOL "True if HDF5 was found by the modern method" FORCE)
```

Comment:
The only suggestion I would have to be to change this to `INTERNAL` as:

```
set(HDF5_FOUND_MODERN_CONFIG_FILE ${INTERNAL_IS_MODERN} CACHE INTERNAL "True if HDF5 was found by the modern method")
```

because we don't want this showing up as an option that a user might be able to change (like with the CMake GUI or with ccmake).